documentation/release_notes.markdown

```diff
@@ -4,6 +4,176 @@ layout: default
 canonical: "/puppetdb/latest/release_notes.html"
 ---
 
+[configure_postgres]: ./configure.html#using-postgresql
+[pg_trgm]: http://www.postgresql.org/docs/current/static/pgtrgm.html
+
+2.3.0
+-----
+
+PuppetDB 2.3.0 is a backwards-compatible release that adds support for
+Puppet 4.
+
+### Upgrading
+
+* For the best-possible performance and scaling capacity, we recommend
+  PostgreSQL version 9.4 or newer with the [`pg_trgm`][pg_trgm]
+  extension enabled, as explained [here][configure_postgres], and we
+  have officially deprecated versions earlier than 9.2.  HSQLDB is
+  only recommended for local development because it has a number of
+  scaling and operational issues.
+
+* Make sure that all of your PuppetDB instances are shut down, and
+  only upgrade one at a time.
+
+* Make sure to upgrade your puppetdb-terminus package (on the host
+  where your Puppet Master lives), and restart your master service.
```

Comment:
I've received confirmation from docs that "Puppet master" is the preferred capitalization. Might as well change it here since this blurb seems to get copy-pasted each time we release.